include/devicetree/reset.h

```diff
@@ -0,0 +1,309 @@
+
+/**
+ * @brief Get a reset specifier's cell value by name
+ *
+ * Example devicetree fragment:
+ *
+ *     reset: reset-controller@... {
+ *             compatible = "vnd,reset";
+ *             #reset-cells = <1>;
+ *     };
+ *
+ *     n: node {
+ *             resets = <&reset 10>;
+ *             reset-names = "alpha";
+ *     };
+ *
+ * Bindings fragment for the vnd,reset compatible:
+ *
+ *     reset-cells:
+ *       - id
+ *
+ * Example usage:
+ *
+ *     DT_RESET_CELL_BY_NAME(DT_NODELABEL(n), alpha, id) // 10
+ *
+ * @param node_id node identifier for a node with a resets property
+ * @param name lowercase-and-underscores name of a resets element
+ *             as defined by the node's reset-names property
+ * @param cell lowercase-and-underscores cell name
+ * @return the cell value in the specifier at the named element
+ * @see DT_PHA_BY_NAME()
+ */
+#define DT_RESET_CELL_BY_NAME(node_id, name, cell) \
+	DT_PHA_BY_NAME(node_id, resets, name, cell)
+
+/**
+ * @brief Equivalent to DT_RESET_CELL_BY_IDX(node_id, 0, cell)
+ * @param node_id node identifier for a node with a resets property
+ * @param cell lowercase-and-underscores cell name
+ * @return the cell value at index 0
+ * @see DT_RESET_CELL_BY_IDX()
+ */
+#define DT_RESET_CELL(node_id, cell) \
+	DT_RESET_CELL_BY_IDX(node_id, 0, cell)
+
+/**
+ * @brief Get the node identifier for the controller phandle from a
+ *        "resets" phandle-array property at an index
+ *
+ * @param inst instance number
+ * @param idx logical index into "resets"
+ * @return the node identifier for the reset controller referenced at
+ *         index "idx"
+ * @see DT_RESET_CTLR_BY_IDX()
+ */
+#define DT_RESET_INST_CTLR_BY_IDX(inst, idx) \
+	DT_RESET_CTLR_BY_IDX(DT_DRV_INST(inst), idx)
+
+/**
+ * @brief Equivalent to DT_RESET_INST_CTLR_BY_IDX(inst, 0)
+ * @param inst instance number
+ * @return a node identifier for the reset controller at index 0
+ *         in "resets"
+ * @see DT_RESET_CTLR()
+ */
+#define DT_RESET_INST_CTLR(inst) \
+	DT_RESET_INST_CTLR_BY_IDX(inst, 0)
+
+/**
+ * @brief Get the node identifier for the controller phandle from a
+ *        resets phandle-array property by name
+ *
+ * @param inst instance number
+ * @param name lowercase-and-underscores name of a resets element
+ *             as defined by the node's reset-names property
+ * @return the node identifier for the reset controller referenced by
+ *         the named element
+ * @see DT_RESET_CTLR_BY_NAME()
+ */
+#define DT_RESET_INST_CTLR_BY_NAME(inst, name) \
+	DT_RESET_CTLR_BY_NAME(DT_DRV_INST(inst), name)
+
+/**
+ * @brief Get a DT_DRV_COMPAT instance's reset specifier's cell value
+ *        at an index
+ * @param inst DT_DRV_COMPAT instance number
+ * @param idx logical index into resets property
+ * @param cell lowercase-and-underscores cell name
+ * @return the cell value at index "idx"
+ * @see DT_RESET_CELL_BY_IDX()
+ */
+#define DT_RESET_INST_CELL_BY_IDX(inst, idx, cell) \
```

Comment:
The rule we follow in our API names is that `DT_INST_FOO(inst, ...)` is the same as `DT_FOO(DT_DRV_INST(inst), ...)`. These macro implementations look OK to me, but the names are not following the pattern.

For example, this should be `DT_INST_RESET_CELL_BY_IDX`, not `DT_RESET_INST_CELL_BY_IDX`.

Please change all of the `DT_RESET_INST_FOO` macros to `DT_INST_RESET_FOO` to follow the rule.

Reply:
I see, I modified this.